books_scraper/spiders/books.py

```diff
@@ -0,0 +1,65 @@
+import scrapy
+from scrapy.http import Response
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/index.html"]
+
+    @staticmethod
+    def parse_detail_info_from_book(response: Response):
+        ratings = {
+            "One": "1",
+            "Two": "2",
+            "Three": "3",
+            "Four": "4",
+            "Five": "5"
+        }
+        available_text = " ".join(
+            response.css("p.instock.availability::text").getall()).strip()
+        available = ""
+
+        for i in available_text:
+            if i.isdigit():
+                available += i
+
+        rating_word = response.css(
+            "p.star-rating::attr(class)"
+        ).get().split()[-1]
+        rating = ratings.get(rating_word, 0)
+
+        yield {
+            "title": response.css("h1::text").get(),
+            "price": response.css(".price_color::text").get().replace("£", ""),
+            "amount_in_stock": available if len(available) > 0 else "0",
+            "rating": rating,
+            "category": response.css(
+                ".breadcrumb > li:nth-last-child(2) a::text"
+            ).get(),
+            "description": response.css(
+                ".product_page > p::text"
+            ).get()[:-7] or "No description",
```

Comment:
The slicing operation `[:-7]` on the description might cause an error if the description is shorter than 7 characters or is `None`. Consider checking if the description exists and has sufficient length before slicing, or handle cases where it might be `None`.